### PR TITLE
[Finetune] Resolve the exception caused by missing validation data and the missing gaudi_config parameter in GaudiTrainer.

### PR DESCRIPTION
**Pull request by minmingzhu**
1. add gaudi_config. When using Gaudi, it will be load gaudi_config.json. One is to download from huggingface with gaudi_config_name and the other is to create new.

image

2. fix validation dataset to be None

### CODE REVIEW

**Comment by xwu99** [comment]
@minmingzhu could you specifically write a clear title for what issue you are fixing?

**Comment by minmingzhu** [comment]
> @minmingzhu could you specifically write a clear title for what issue you are fixing?

Resolved

**Comment by minmingzhu** [comment]
> Resolved



> @minmingzhu could you specifically write a clear title for what issue you are fixing?

Resolved

**Comment by minmingzhu** [comment]
@xwu99 
Resolved

File: llm_on_ray/finetune/finetune.py [hunk 1]
```diff
@@ -271,13 +273,20 @@ def train_func(config: Dict[str, Any]):
     elif device in ["hpu"]:
         from optimum.habana.transformers import GaudiTrainer
         from optimum.habana.transformers import GaudiTrainingArguments
+        from optimum.habana import GaudiConfig
 
+        gaudi_config = GaudiConfig()
+        gaudi_config.use_fused_adam = True
```
Comment (harborn) [inline llm_on_ray/finetune/finetune.py:279]: Maybe should add a option in yaml config to use fused_adam on HPU fine-tuning.
fused_adam is provided in optimum-habana, this is only for HPU fine-tuning.

**Review by harborn — APPROVED** [review]

**Pull request merged**